models/transformer.py

```diff
@@ -56,6 +56,8 @@ def forward(self, src, mask, query_embed, pos_embed):
         memory = self.encoder(src, src_key_padding_mask=mask, pos=pos_embed)
         hs = self.decoder(tgt, memory, memory_key_padding_mask=mask,
                           pos=pos_embed, query_pos=query_embed)
+        if not self.decoder.return_intermediate:
+            hs = hs.unsqueeze(0)
```

Comment:
Can you move this instead to https://github.com/facebookresearch/detr/blob/f4cdc542de34de771da8b9189742e5465f5220cd/models/transformer.py#L124 so that the code in `TransformerDecoder` returns a consistent tensor dimensionality?

Reply:
Hi @fmassa, I moved the `unsqueeze(0)` into the TransformerDecoder class